src/data/markdown/docs/02 javascript api/07 k6-experimental/04 tracing/01 instrumentHTTP.md

```diff
@@ -0,0 +1,45 @@
+---
+title: 'instrumentHTTP'
+excerpt: 'instrumentHTTP instruments the k6 http module with tracing capabilities.'
+---
+
+The `instrumentHTTP` function instruments the k6 http module with tracing capabilities. It will transparently replace each of the k6 http module [del](/javascript-api/k6-http/del), [get](/javascript-api/k6-http/get), [head](/javascript-api/k6-http/head), [options](/javascript-api/k6-http/options), [patch](/javascript-api/k6-http/patch), [post](/javascript-api/k6-http/post), [put](/javascript-api/k6-http/head), and [request](/javascript-api/k6-http/request) functions with versions that will automatically attach a trace context to every request.
+
+From a user perspective, the `instrumentHTTP` allows automatically adding tracing information to HTTP requests without having to rewrite any existing code. Calling it once in the init context will ensure that all requests made by the HTTP module from that point forward will have a trace context header added to them and will add the used `trace_id` as part of the produced data points output metadata.
+
+The `instrumentHTTP` function takes a single configuration object argument that can be used to configure the tracer. The argument should be an instance of the [`Options`](/javascript-api/k6-experimental/tracing/options) object.
+
+## Example
+
+This example demonstrates how calling the `instrumentHTTP` function in the init context of a script once will ensure that all requests made by the HTTP module from that point forward will have a trace context header attached.
```

Comment:
```suggestion
This example demonstrates how calling the `instrumentHTTP` function in the init context of a script once ensures that all requests made by the HTTP module from that point forward will have a trace context header attached.
```

Reply:
I don't exactly understand what the `propagator: "w3c"` argument does here.  Can you explain?

Reply:
I see it says configure the tracer, but I'm not sure what it does. Maybe the intro could provide one sentence of context.

I'm guessing the trace-context header is `"X-Example-Header": "instrumented/get"`, is that correct?

Reply:
And one more thing, I don't see where the aforementioned trace-id is .

Reply:
That's a really good question. I guess this requires some context that we need to clarify indeed ✋🏻
Tracing in k6 consists of generating a trace-id (an alphanumeric sequence that's somewhat random), and attaching it to each HTTP request as a "trace-context header". There are different specifications for what that header should look like (the key it uses and its value). Attaching a trace id to a request is called "propagating" it (because it is passed from service to service).

As a result, the propagator is the "propagation format", meaning the name of the propagation (trace context format) specification used when generating and attaching a trace id to each request.

Please don't hesitate to let me know if that makes no sense or if you need further clarification. I realize now that I could make this lower context and not assume that all users will be super comfortable with tracing.

I'll give a shot at expanding a bit on this 👍🏻 🙇🏻

Reply:
I will add:
- [ ] A description of what a "propagator" is
- [ ] A description of what a "trace id" is
- [ ] Document what headers are to be expected according to the selected propagator

Reply:
I've attempted to address it in https://github.com/grafana/k6-docs/pull/1020/commits/ec13abe71ae78b068baaac7cefb4deead8b39445 @MattDodsonEnglish. Let me know what you think, and once we've reached a state we're happy with, I'll add it to the client documentation too 👍🏻